src/osd/PrimaryLogPG.cc

```diff
@@ -5862,8 +5871,9 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	  break;
 	}
 
-	if (op.extent.offset > cct->_conf->osd_max_object_size) {
-	  result = -EFBIG;
+	if (op.extent.offset > osd_max_object_size) {
```

Comment:
why do this check twice? we already know that the the op will end up with an over-sized object.

Reply:
@tchaikov Well, this block is already there. I don't think I added it newly. See:
https://github.com/ceph/ceph/blob/master/src/osd/PrimaryLogPG.cc#L5865-L5868

Reply:
i mean, you could drop this line, if you choose to use the helper function. otherwise, we do the check twice **after this change**.